client/internal/receivedprocessor.go

```diff
@@ -32,6 +35,7 @@ func newReceivedProcessor(
 	logger types.Logger,
 	callbacks types.Callbacks,
 	sender Sender,
+	senderCommon *SenderCommon,
```

Comment:
I agree it's confusing but since Sender is an interface I thought SenderCommon is the relevant struct for adding logic the user won't need to implement. Perhaps have Sender hold a SenderCommon instance?

Reply:
>Perhaps have Sender hold a SenderCommon instance?

This doesn't sound right. SenderCommon is an implementation, Sender is an interface.

Reply:
Ok thanks, please see the latest commit if I understood correctly

Reply:
Yes, that's what I was envisioning. Thank you.